pkgs/build-support/php/build-composer-repository.nix

```diff
@@ -0,0 +1,79 @@
+{ stdenvNoCC, lib, writeTextDir, fetchFromGitHub, php, composer-local-repo-plugin }:
+
+let
+  mkComposerRepositoryOverride =
+    /*
+      We cannot destruct finalAttrs since the attrset below is used to construct it
+      and Nix currently does not support lazy attribute names.
+      {
+      php ? null,
+      composer ? null,
+      composerLock ? "composer.lock",
+      src,
+      vendorHash,
+      ...
+      }@finalAttrs:
+    */
+    finalAttrs: previousAttrs:
+
+    let
+      phpDrv = finalAttrs.php or php;
+      composer = finalAttrs.composer or php.packages.composer;
```

Comment:
This should use `phpDrv.packages.composer` instead of `php.packages.composer`. As currently composer will be the wrong php version or be missing all extra configuration inside `finalAttrs.php` that a packager decides on. Leading to also require setting up a redundant line, `composer = phpDrv.packages.composer` on the package level.

Reply:
Just fixed now. Thanks!